app/modules/conversations/access/access_service.py

```
@@ -38,49 +38,37 @@ async def share_chat(
-            )
-            self.db.commit()
 
-        return conversation_id
+        except IntegrityError as e:
+            self.db.rollback()
+            raise ShareChatServiceError(
+                "Failed to update shared chat due to a database integrity error."
+            ) from e
+        except Exception as e:
+            self.db.rollback()
+            raise ShareChatServiceError(f"Failed to update shared chat: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)